.github/workflows/build-user-config.yml

```
@@ -0,0 +1,124 @@
+name: Reusable user config build
+
+on: 
+  workflow_call:
+    inputs:
+      build_matrix_path:
+        description: 'Path to the build matrix file'
+        default: 'build.yaml'
+        required: false
+        type: string
+      config_path:
+        description: 'Path to the config directory'
+        default: 'config'
+        required: false
+        type: string
+      fallback_binary:
+        description: 'Fallback binary format, if no *.uf2 file was built'
+        default: 'bin'
+        required: false
+        type: string
+
+jobs:
+  matrix:
+    runs-on: ubuntu-latest
+    name: Fetch Build Keyboards
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install yaml2json
+        run: python3 -m pip install remarshal
+
+      - id: set-matrix
+        name: Fetch Build Matrix
+        run: |
+          matrix=$(yaml2json ${{ inputs.build_matrix_path }} | jq -c .)
+          yaml2json ${{ inputs.build_matrix_path }}
+          echo "::set-output name=matrix::${matrix}"
+
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: zmkfirmware/zmk-build-arm:stable
+    needs: matrix
+    name: Build
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.matrix.outputs.matrix)}}
+    steps:
+      - name: Prepare variables
+        id: variables
+        run: |
+          if [ -n "${{ matrix.shield }}" ]; then
+            EXTRA_CMAKE_ARGS="-DSHIELD=${{ matrix.shield }}"
+            ARTIFACT_NAME="${{ matrix.shield }}-${{ matrix.board }}-zmk"
+            DISPLAY_NAME="${{ matrix.shield }} - ${{ matrix.board }}"
+          else
+            EXTRA_CMAKE_ARGS=
+            DISPLAY_NAME="${{ matrix.board }}"
+            ARTIFACT_NAME="${{ matrix.board }}-zmk"
+          fi
+          echo ::set-output name=extra-cmake-args::${EXTRA_CMAKE_ARGS}
+          echo ::set-output name=artifact-name::${ARTIFACT_NAME}
+          echo ::set-output name=display-name::${DISPLAY_NAME}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Cache west modules
+        uses: actions/cache@v2
```

```suggestion
        uses: actions/cache@v3.0.1
```
Align this change with #1204

---

Updated